### PR TITLE
test for symlinks to being chased

### LINKED ISSUE #4344: symlink dirs: add "realpath" test case
At present `symlink dirs` does not attempt to resolve the path when it creates a symlink.

This can be a useful feature, for example if a site has redundancy in its filesystem they might choose to put whole filesystem trees behind a symlink allowing them to switch source in the event of failure.

This use case was contributed in metomi-rose as Rose 2019 used to resolve paths for its `root-dir` functionality https://github.com/metomi/rose/issues/2131.

We should add a test case to ensure that this functionality is preserved.

Example to test:

* Ceate a directory and a symlink pointing at that directory (e.g. `~/foo -> foo1`).
* Configure `symlink dirs` to use the symlink (e.g. `--symlink-dirs=run=~/foo`).
* Any symlinks created by Cylc should *not* be pointing at the resolved path (e.g. should be `~/foo/...` rather than `~/foo1/...`).

> **Note:** This really applies to all symlinks not just `symlink dirs`, it might be a good idea to squeeze in a quick symlink scan here too.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #4344 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (testing chage).
- [x] No documentation update required.

### CODE REVIEW

File: tests/functional/cylc-install/04-symlinks-not-resolved.t [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# Install the workflow:
+run_ok "$TEST_NAME_BASE" cylc install --no-run-name --directory "$PWD" --symlink-dirs=run="${elsewhere}/bar"
+
+# Check the installed workflow:
+ls -l "$RUN_DIR/$(basename "$PWD")" > list
+grep_ok "bar\/cylc-run" list
+
+# Tidy up:
+cylc clean "$(basename "$PWD")" 2> /dev/null
```
Comment (datamel) [inline tests/functional/cylc-install/04-symlinks-not-resolved.t:49]: Probably a long shot, does this clean have the potential to mess with future tests running in parallel? 
Wondering if use of `make_rnd_workflow` and `purge_rnd_workflow` from test header would be safer, but not a deal breaker though.

File: tests/functional/cylc-install/04-symlinks-not-resolved.t [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[scheduling]
+    initial cycle point = 1500
+    [[graph]]
+        R1 = foo
+__HEREDOC__
+
+run_ok "cylc validate ."
+
+# Create a temporary directory to put our symlinked foo and bar in:
+elsewhere=$(mktemp -d)
```
Comment (datamel) [inline tests/functional/cylc-install/04-symlinks-not-resolved.t:37]: My bash inexperience here so happy to be corrected, should this `elsewhere` be in capitals?

**Review by datamel — APPROVED** [review]
Looks good, one minor comment and one question.

File: tests/functional/cylc-install/04-symlinks-not-resolved.t (hunk 2 above)
Comment (wxtim) [reply tests/functional/cylc-install/04-symlinks-not-resolved.t:37]: Bash doesn't mandate it. It's just common practice.

File: tests/functional/cylc-install/04-symlinks-not-resolved.t (hunk 1 above)
Comment (wxtim) [reply tests/functional/cylc-install/04-symlinks-not-resolved.t:49]: What's the benefit? I've not come across `make/purge_rnd_workflow?`

**Review by kinow — APPROVED** [review]

File: tests/functional/cylc-install/04-symlinks-not-resolved.t (hunk 1 above)
Comment (datamel) [reply tests/functional/cylc-install/04-symlinks-not-resolved.t:49]: It just makes/deletes a random source directory with a `flow.cylc` in it, which would remove the need for your `flow.cylc` at the top too.

**Pull request merged**